paramak/utils.py

```diff
@@ -271,15 +282,44 @@ def get_hash(shape, ignored_keys=[]):
     hash_object = blake2b()
     shape_dict = dict(shape.__dict__)
 
-    for key in ignored_keys:
-        if key in shape_dict.keys():
-            shape_dict[key] = None
+    if ignored_keys is not None:
+        for key in ignored_keys:
+            if key in shape_dict.keys():
+                shape_dict[key] = None
 
     hash_object.update(str(list(shape_dict.values())).encode("utf-8"))
     value = hash_object.hexdigest()
     return value
 
 
+def _replace(filename: str, pattern: str, subst: str) -> None:
+    """Opens a file and replaces occurances of a particular string
+        (pattern)with a new string (subst) and overwrites the file.
+        Used internally within the paramak to ensure .STP files are
+        in units of cm not the default mm.
+    Args:
+        filename (str): the filename of the file to edit
+        pattern (str): the string that should be removed
+        subst (str): the string that should be used in the place of the
+            pattern string
+    """
+    # Create temp file
+    file_handle, abs_path = mkstemp()
+    with fdopen(file_handle, 'w') as new_file:
+        with open(filename) as old_file:
+            for line in old_file:
+                new_file.write(line.replace(pattern, subst))
+
+    # Copy the file permissions from the old file to the new file
+    copymode(filename, abs_path)
+
+    # Remove original file
+    remove(filename)
+
+    # Move new file
+    move(abs_path, filename)
```

Comment:
This utility function isn't tested which makes the diff coverage drop

Reply:
an interesting, I had just moved that from one of the classes to the utils functions. I can added tests